### PR TITLE
Rename ksql-cli to ksql. Move standalone mode to server execution

### PR DESCRIPTION
**Pull request by dguy**
This PR addresses both #774 and #773.

1. `ksql-cli` has been renamed to `ksql`.
2. `ksql` only has a single "mode", i.e., it is always remote
3.  the server can now be executed with `--queries-file` to execute it "standalone mode"

### CODE REVIEW

File: ksql-cli/src/main/java/io/confluent/ksql/Ksql.java [hunk 1]
```diff
@@ -16,42 +16,64 @@
...
-        System.err.println(exception.getMessage());
-      } else {
-        System.err.println("Options parsing failed for an unknown reason");
-      }
-      System.err.println("See the help command for usage information");
-    } catch (Exception e) {
-      System.err.println(CommonUtils.getErrorMessageWithCause(e));
+  public static void main(String[] args) throws IOException {
+    final Options options = Options.parse(args);
+    if (options == null) {
```
Comment (hjafarpour) [inline ksql-cli/src/main/java/io/confluent/ksql/Ksql.java]: Should we print an error message before exit?

File: ksql-cli/src/main/java/io/confluent/ksql/Ksql.java [hunk 2]
```diff
@@ -16,42 +16,64 @@
...
+        restClient,
+        new JLineTerminal(options.getOutputFormat(), restClient))) {
+        cli.runInteractively();
     }
+  }
 
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private static Properties loadProperties(final Optional<String> propertiesFile) {
+    final Properties properties = new Properties();
+    properties.put(StreamsConfig.APPLICATION_ID_CONFIG, KsqlConfig.KSQL_SERVICE_ID_DEFAULT);
```
Comment (hjafarpour) [inline ksql-cli/src/main/java/io/confluent/ksql/Ksql.java]: `StreamConfig. APPLICATION_ID_CONFIG` will be set by KSQL so no need to have it anymore.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/ServerMain.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private static Executable createExecutable(final Properties properties,
+                                             final Optional<String> queriesFile) throws Exception {
+    if (queriesFile.isPresent()) {
+      return StandaloneExecutor.create(properties, queriesFile.get());
+    }
+
+    if (!properties.containsKey(StreamsConfig.APPLICATION_ID_CONFIG)) {
```
Comment (hjafarpour) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/ServerMain.java]: Same as above, no need for this.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    shutdownLatch.countDown();
+  }
+
+  @Override
+  public void join() throws InterruptedException {
+    shutdownLatch.await();
+  }
+
+  public static StandaloneExecutor create(final Properties properties, final String queriesFile) {
+    if(!properties.containsKey(StreamsConfig.APPLICATION_ID_CONFIG)) {
```
Comment (hjafarpour) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java]: Ditto.

**Comment by blueedgenick** [comment]
_(originally made this comment on #773 directly but this seems a better place for it)_
i think that magically disabling the http listener is too extreme if there's no way to turn it back on again. Feedback suggests that the ability to keep the http listener alive, even if only on one server node out of a group which is processing the same sql application, is a useful and desirable capability for sysadmins. The folks I've talked to all suggested some variant of using e.g. nginx reverse proxy to limit access to that one node but they do want the ability to log in via a cli and "poke around" to check status etc.
Therefore I suggest that we move to more explicit requesting of behavior with these parameters - perhaps introduce an extra flag for --listen-enabled or similar to give people the option of rolling their own permutations of whether or not the http endpoint is enabled and/or a query file will be processed.

File: ksql-cli/src/main/java/io/confluent/ksql/Ksql.java (hunk 1 above)
Comment (dguy) [reply ksql-cli/src/main/java/io/confluent/ksql/Ksql.java]: It gets printed during parsing

File: ksql-cli/src/main/java/io/confluent/ksql/Ksql.java (hunk 2 above)
Comment (dguy) [reply ksql-cli/src/main/java/io/confluent/ksql/Ksql.java]: where?

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/ServerMain.java (hunk 3 above)
Comment (dguy) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/ServerMain.java]: It fails if the application.id isn't set here and not set in the properties file

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java (hunk 4 above)
Comment (dguy) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java]: as above

**Comment by dguy** [comment]
@miguno

File: ksql-cli/src/main/java/io/confluent/ksql/Ksql.java (hunk 2 above)
Comment (hjafarpour) [reply ksql-cli/src/main/java/io/confluent/ksql/Ksql.java]: It's set in `PhysicalPlanBuilder.java`.

**Comment by apurvam** [comment]
One thing I would mention in the context of having an open http port with standalone mode, this is actually a new feature: the current standalone mode doesn't have a port open. I don't think we should do it in this patch. 

However, it would be helpful to print the log locations of the standalone app on the console, to make it that much easier for people to go look at what's going on under the hood.

**Pull request merged**